dialects/affine/passes/lower_for.cpp

```diff
@@ -35,18 +35,20 @@ const Def* LowerFor::rewrite(const Def* def) {
         { // construct for
             auto [iter, end, step, acc] = for_lam->vars<4>();
 
+
+
             // reduce the body to remove the cn parameter
             auto nom_body = body->as_nom<Lam>();
-            auto new_body = nom_body->stub(w, w.cn(w.sigma()), body->dbg());
-            new_body->set(nom_body->reduce(w.tuple({iter, acc, yield_lam})));
+            auto new_body = nom_body->stub(w, w.cn(acc->type()), body->dbg())->as<Lam>();
+            new_body->set(nom_body->reduce(w.tuple({iter, new_body->var(), yield_lam})));
 
             // break
-            auto if_else_cn = w.cn(w.sigma());
+            auto if_else_cn = w.cn(acc->type());
             auto if_else    = w.nom_lam(if_else_cn, nullptr);
-            if_else->app(false, brk, acc);
+            if_else->app(false, brk, if_else->var());
 
             auto cmp = core::op(core::icmp::ul, iter, end);
-            for_lam->branch(false, cmp, new_body, if_else, w.tuple());
+            for_lam->branch(false, cmp, new_body, if_else, acc);
```

Comment:
@christopherhjung 
Is the inlining/forwarding of the accumulator necessary here?
Optimizations like CopyProp will probably delete the arguments.
And Closure Conversion has as one main function closing open terms and will (if not inlined) re-add the accumulator and bound variables as arguments.